packages/shared-ux/button_toolbar/src/buttons/icon_button_group/icon_button_group.styles.ts

```diff
@@ -14,11 +14,13 @@ export const IconButtonGroupStyles = ({ euiTheme }: UseEuiTheme) => {
       '&.euiButtonGroupButton': {
         backgroundColor: euiTheme.colors.emptyShade,
         border: `${euiTheme.border.thin} !important`,
+        borderRight: 'none !important',
         '&:first-of-type': {
           borderTopLeftRadius: `${euiTheme.border.radius.medium} !important`,
           borderBottomLeftRadius: `${euiTheme.border.radius.medium} !important`,
         },
         '&:last-of-type': {
+          borderRight: `${euiTheme.border.thin} !important`,
```

Comment:
These style changes fix the old "double border" between each of the icon buttons in the `IconButtonGroup`:

![CompareIconGroup](https://user-images.githubusercontent.com/8698078/220712923-dcaeebab-67b8-46cc-bb5d-62b70cbe1297.png)